akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -75,6 +78,22 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
     activeStreamCount() < maxConcurrentStreams
   }
 
+  private var completing = false
+
+  /**
+   * Marks this object to be shutting down. If the buffer is empty, it also invokes [[onComplete()]].
```

Comment:
Nitpick: A bit superflous repeating exactly what is in the code with the "if a then calls b"?